gobblin-runtime/src/main/java/gobblin/runtime/Task.java

```diff
@@ -112,9 +120,10 @@ public void run() {
     Closer closer = Closer.create();
     try {
       // Build the extractor for extracting source schema and data records
-      Extractor extractor = closer.register(new ExtractorDecorator(
+      Extractor extractor = closer.register(new InstrumentedExtractorDecorator(this.taskState,
```

Comment:
Yes, the current design will do that. I don't think it should be too much overhead, but we reevaluate if we see it is.